aodncore/pipeline/steps/upload.py

```diff
@@ -221,7 +221,7 @@ def _get_absolute_dest_uri(self, pipeline_file):
     @retry_decorator(**retry_kwargs)
     def _get_is_overwrite(self, pipeline_file, abs_path):
         response = self.s3_client.list_objects_v2(Bucket=self.bucket, Prefix=abs_path)
-        return bool(k for k in response.get('Contents', []) if k['Key'] == abs_path)
+        return bool([k for k in response.get('Contents', []) if k['Key'] == abs_path])
```

Comment:
Just for reference purposes, the reason this was bugged was because every generator is "True" even if it is "empty", e.g.:

```python
In [1]: l = []

In [2]: g = (e for e in l)

In [3]: g
Out[3]: <generator object <genexpr> at 0x7f752253f410>

In [4]: bool(g)
Out[4]: True

In [5]: g.next()
---------------------------------------------------------------------------
StopIteration                             Traceback (most recent call last)
<ipython-input-5-7dbbdfed0980> in <module>()
----> 1 g.next()

StopIteration:
```

Reply:
Sweet, thanks @lwgordonimos